tests/unit/consensus/test_base.py

```diff
@@ -1460,15 +1475,295 @@ def get_vote():
+        Appeal leader succeeds: leader disagrees + 22 validators disagree for 2 rounds then agree for 1 round
+
+        Appeal validator succeeds: 25 validators disagree
+        Leader disagrees + 46 validators disagree. -> 47 rounds
+        Appeal leader fails: leader disagrees + 94 validators disagree. -> 95 rounds
+        """
+        n_first = DEFAULT_VALIDATORS_COUNT
+        n_second = 2 * n_first + 1
+        n_third = 2 * n_second + 1
+        nb_first_agree = (n_first**2) + (n_second**2) + (n_third * 2)
+        if (len(created_nodes) >= nb_first_agree) and (
+            len(created_nodes) < nb_first_agree + n_third
+        ):
+            return Vote.AGREE
+        else:
+            return Vote.DISAGREE
+
+    transactions_processor = TransactionsProcessorMock(
+        [transaction_to_dict(transaction)]
+    )
+
+    msg_handler_mock = Mock(MessageHandler)
+
+    node_factory_supplier = (
+        lambda node, mode, contract_snapshot, receipt, msg_handler, contract_snapshot_factory: created_nodes.append(
+            node_factory(
+                node,
+                mode,
+                contract_snapshot,
+                receipt,
+                msg_handler,
+                contract_snapshot_factory,
+                get_vote(),
+            )
+        )
+        or created_nodes[-1]
+    )
+
+    managed_thread(
+        transactions_processor, msg_handler_mock, nodes, node_factory_supplier
+    )
+
+    await ConsensusAlgorithm(None, msg_handler_mock).exec_transaction(
+        transaction=transaction,
+        transactions_processor=transactions_processor,
+        snapshot=SnapshotMock(nodes),
+        accounts_manager=AccountsManagerMock(),
+        contract_snapshot_factory=contract_snapshot_factory,
+        node_factory=node_factory_supplier,
+    )
+
+    for node in created_nodes:
+        node.exec_transaction.assert_awaited_once_with(transaction)
+
+    assert (
+        transactions_processor.get_transaction_by_hash(transaction.hash)["status"]
+        == TransactionStatus.UNDETERMINED.value
+    )
+
+    transaction_status_history = [
+        *[
+            TransactionStatus.PROPOSING,
+            TransactionStatus.COMMITTING,
+            TransactionStatus.REVEALING,
+        ]
+        * DEFAULT_VALIDATORS_COUNT,
+        TransactionStatus.UNDETERMINED,
+    ]
+    assert transactions_processor.updated_transaction_status_history == {
+        "transaction_hash": transaction_status_history
+    }
+
+    nb_validators = DEFAULT_VALIDATORS_COUNT
+    nb_created_nodes = DEFAULT_VALIDATORS_COUNT**2
+    check_validator_count(transaction, transactions_processor, nb_validators)
+    assert len(created_nodes) == nb_created_nodes
+
+    appeal(transaction, transactions_processor)
+    time.sleep(3)
+
+    transaction = Transaction.from_dict(
+        transactions_processor.get_transaction_by_hash(transaction.hash)
+    )
+    await ConsensusAlgorithm(None, msg_handler_mock).exec_transaction(
```

Comment:
@kstroobants this seems like a different test, shouldn't we divide them into different ones?

Reply:
I agree that the test could have been split into an appeal fail test and an appeal success test. I wanted to also run them together to check the dependencies (i.e the correct amount of validators selected etc.). Hence, I did not see a benefit of duplicating them.